fractal_tasks_core/lib_masked_loading.py

```diff
@@ -0,0 +1,248 @@
+        f"{current_label_array.shape=}"
+    )
+
+    # Load ROI data for current label array
+    current_label_region = current_label_array[region].compute()
+
+    # Load ROI data for masking label array, with or without upscaling
+    if masking_label_array.shape != current_label_array.shape:
+        logger.info("Upscaling of masking label is needed")
+        lowres_region = convert_region_to_low_res(
+            highres_region=region,
+            highres_shape=current_label_array.shape,
+            lowres_shape=masking_label_array.shape,
+        )
+        masking_label_region = masking_label_array[lowres_region].compute()
+        masking_label_region = upscale_array(
+            array=masking_label_region,
+            target_shape=current_label_region.shape,
+        )
+    else:
+        masking_label_region = masking_label_array[region].compute()
+
+    # Check that all shapes match
+    shapes = (
+        masking_label_region.shape,
+        current_label_region.shape,
+        image_array.shape[1:],
+    )
+    if len(set(shapes)) > 1:
+        raise ValueError(
+            "Shape mismatch:\n"
+            f"{current_label_region.shape=}\n"
+            f"{masking_label_region.shape=}\n"
+            f"{image_array.shape=}"
+        )
+
+    # Compute background mask
+    background_3D = masking_label_region != label_value
+    if (masking_label_region == label_value).sum() == 0:
+        raise ValueError(
+            f"Label {label_value} is not present in the extracted ROI"
+        )
+
+    # Set image background to zero
+    background_4D = np.expand_dims(background_3D, axis=0)
+    image_array[background_4D] = 0
+
+    return (image_array, background_3D, current_label_region)
+
+
+def postprocess_cellpose_output(
+    *,
+    modified_array: np.ndarray,
+    original_array: np.ndarray,
+    background: np.ndarray,
+) -> np.ndarray:
+    """
+    Postprocess a cellpose input, mainly to restore its original background
+
+    **FIXME 1**: review/improve variable names
+
+    **FIXME 2**: make this function more flexible, and move it to a separate
+    module (see `issue 340
+    <https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/340>`_).
+
+    :param modified_array: The 3D (ZYX) array with the correct object data and
+                           wrong background data.
+    :param original_array: The 3D (ZYX) array with the wrong object data and
+                           correct background data.
+    :param background: The 3D (ZYX) boolean array that defines the
+                       background.
+    """
+
+    # Restore background
+    modified_array[background] = original_array[background]
+    return modified_array
+
+
+def wrap_function_with_masking(
+    *,
+    function: Callable,
+    image_array: np.ndarray,
+    kwargs: Optional[dict] = None,
+    use_masks: bool,
+    preprocessing_kwargs: Optional[dict] = None,
+    postprocessing_kwargs: Optional[dict] = None,
```

Comment:
Do we expect to have postprocessing kwargs? If yes, we'd need to also pass them to the postprocessing function. Otherwise, we should remove the parameter

Reply:
Agreed.
In the current version, there are no postprocessing kwargs.